ui/shared/AccountSelector.qml

```diff
@@ -22,10 +20,38 @@ Item {
     property string showAssetBalance: ""
     property int dropdownWidth: width
     property alias dropdownAlignment: select.menuAlignment
+    property bool isValid: true
+    property var reset: function() {}
+
+    function resetInternal() {
+        accounts = undefined
+        selectedAccount = undefined
+        isValid = true
+    }
+
+    onSelectedAccountChanged: {
+        if (selectedAccount) {
+            if (selectedAccount.iconColor) {
+                selectedIconImgOverlay.color = selectedAccount.iconColor
+            }
+            if (selectedAccount.name) {
+                selectedTextField.text = selectedAccount.name
+            }
+            if (selectedAccount.address) {
+                textSelectedAddress.text = selectedAccount.address  + " • "
+            }
+            if (selectedAccount.fiatBalance) {
+                textSelectedAddressFiatBalance.text = selectedAccount.fiatBalance + " " + currency.toUpperCase()
+            }
+            if (selectedAccount.assets) {
+                rptAccounts.model = selectedAccount.assets
+            }
```

Comment:
Wouldn't it be enough to just check for `selectedAccount` and always update all other properties? Or is there a possibility that `selectedAccount` is updated without any of those props having changed?

Reply:
`selectedAccount` can be specified by parent components and can have values that produce errors in QML, especially when they are `undefined`. For example, setting a `.text` property to `undefined` causes an error.